collect/cache/cuckooSentCache.go

```diff
@@ -0,0 +1,179 @@
+package cache
+
+import (
+	"time"
+
+	lru "github.com/hashicorp/golang-lru"
+	"github.com/honeycombio/refinery/config"
+	"github.com/honeycombio/refinery/types"
+)
+
+// cuckooSentCache extends Refinery's legacy cache. It keeps the same records
+// for kept traces but adds a pair of cuckoo filters to record dropped traces.
+// This allows many more traces to be kept in the cache; now only kept records
+// are retained in the cache of sentRecords.
+// The size of the sent cache is still set based on the size of the live trace cache,
+// and the size of the dropped cache is an independent value.
+
+// cuckooKeptRecord is an internal record we leave behind when keeping a trace to remember
+// our decision for the future. We only store them if the record was kept.
+type cuckooKeptRecord struct {
+	rate      uint // sample rate used when sending the trace
+	spanCount uint // number of spans in the trace (we decorate the root span with this)
+}
+
+func (t *cuckooKeptRecord) Kept() bool {
+	return true
+}
+
+func (t *cuckooKeptRecord) Rate() uint {
+	return t.rate
+}
+
+func (t *cuckooKeptRecord) DescendantCount() uint {
+	return uint(t.spanCount)
+}
+
+func (t *cuckooKeptRecord) Count(*types.Span) {
+	t.spanCount++
+}
+
+// Make sure it implements TraceSentRecord
+var _ TraceSentRecord = (*cuckooKeptRecord)(nil)
+
+// cuckooSentRecord is what a TraceSentRecord we return when the trace was dropped.
+// It's always the same one.
+type cuckooDroppedRecord struct{}
+
+func (t *cuckooDroppedRecord) Kept() bool {
+	return true
```

Comment:
Dropped record should return false, right?

```suggestion
	return false
```